frontend/src/components/login/page/LoginPage.tsx

```diff
@@ -1,7 +1,114 @@
+import "./LoginPageStyle.css";
+import { useSession } from "../../../context/SessionContext";
+import useModal from "../../../hooks/useModal";
 
 const Login: FC = () => {
-  return <div>Loginpage</div>;
+  const { login } = useSession();
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [title, setTitle] = useState("");
+  const [bodyMessage, setBodyMessage] = useState("");
+  const { isShowing, toggle } = useModal(false);
+
+  const handleErrorMessage = useCallback(
+    (title: string, bodyMessage: string) => {
+      setTitle(title);
+      setBodyMessage(bodyMessage);
+      toggle();
+    },
+    [toggle]
+  );
+
+  const handleSubmit = useCallback(
+    async (event) => {
+      event.preventDefault();
+
+      if (username.length <= 5)
+        return handleErrorMessage(
+          "กรุณาใส่ชื่อผู้ใช้งาน",
+          "ชื่อผู้ใช้งานต้องมีอย่างน้อย 6 ตัวอักษร"
+        );
+
+      // if (password.length <= 7)
+      //   return handleErrorMessage("กรุณาใส่รหัสผ่าน", "รหัสผ่านต้องมีอย่างน้อย 8 ตัวอักษร");
+
+      try {
+        await login(username, password);
+      } catch ({ message }) {
+        console.log(message);
+        if (message === "Username not found")
+          return handleErrorMessage("เข้าสู่ระบบไม่สำเร็จ", "ไม่พบบัญชีในระบบ");
+
+        if (message === "Incorrect password")
+          return handleErrorMessage("เข้าสู่ระบบไม่สำเร็จ", "ชื่อผู้ใช้งานหรือรหัสผ่านไม่ถูกต้อง");
+      }
+    },
+    [handleErrorMessage, login, password, username]
+  );
+
+  const handleCallBack = (stats: boolean) => {
+    if (!stats) toggle();
+  };
+
+  return (
+    <div>
+      <Modal
+        isOpen={isShowing}
+        isHasAccept={false}
+        isHasDecline={false}
+        title={title}
+        bodyMessage={bodyMessage}
+        callBackFunction={handleCallBack}
+      />
+      <div className="login-page flex justify-center items-center">
+        <div className="login-box relative p-10">
+          <form onSubmit={handleSubmit}>
+            <p className="text-2xl">เข้าสู่ระบบ</p>
+            <p className="text-xs mt-1">กรอกชื่อผู้ใช้งานและรหัสผ่านเพื่อเข้าใช้งาน</p>
+
+            <div className="mt-16">
+              <p className="text-md">ชื่อผู้ใช้งาน</p>
+              <input
+                onChange={(event) => setUsername(event.target.value)}
+                className="form-input w-full bg-white-100 px-0.5 border-0 border-b-2 border-gray-200 focus:ring-0 focus:border-dark-200"
+                type="text"
+                placeholder="ชื่อผู้ใช้งาน"
+              />
+              <p className="mt-5 text-md">รหัสผ่าน</p>
+              <input
+                onChange={(event) => setPassword(event.target.value)}
+                className="form-input w-full bg-white-100 px-0.5 border-0 border-b-2 border-gray-200 focus:ring-0 focus:border-dark-200"
+                type="password"
+                placeholder="รหัสผ่าน"
+              />
```

Comment:
Should we use `useCallback` to handle input change, since we don't need to render component every time when username or password change, or not?

Reply:
@Icyscools Roger that